clients/src/main/java/org/apache/kafka/common/network/SSLTransportLayer.java

```diff
@@ -222,11 +222,13 @@ public void handshake() throws IOException {
                               channelId, appReadBuffer.position(), netReadBuffer.position(), netWriteBuffer.position());
                     handshakeResult = handshakeWrap(write);
                     if (handshakeResult.getStatus() == Status.BUFFER_OVERFLOW) {
-                        int currentPacketBufferSize = packetBufferSize();
-                        netWriteBuffer = Utils.ensureCapacity(netWriteBuffer, currentPacketBufferSize);
-                        if (netWriteBuffer.position() >= currentPacketBufferSize) {
+                        int currentNetWriteBufferSize = netWriteBufferSize();
+                        netWriteBuffer.position(netWriteBuffer.limit());
```

Comment:
`netWriteBuffer` is flipped inside `handshakeWrap()`, so calls to `flush()` assume that the buffer is already in flipped state, ready to be written to socket. Hence when the buffer is expanded, the buffer needs to be left in the same state.